gapic/src/main/com/google/gapid/models/Profile.java

```diff
@@ -412,6 +351,49 @@ public boolean hasChildren() {
     public List<PerfNode> getChildren() {
       return children;
     }
+
+    protected void addChild(PerfNode node) {
+      children.add(node);
+    }
+
+    protected PerfNode findNode(Path.Commands link, int level) {
+      int idx = Collections.binarySearch(children, null, (node, $) -> {
```

Comment:
Lots of implicit magic going on here too. Can you add some explanation of the data format or something that would help a reader understand how this logic finds perf nodes?

Reply:
Also what is the level argument? Is this the sub command index depth? Why is one of those needed when finding perf nodes for command ranges? How does this function differ from findNodeContaining below which appears to do the same thing but lacks an index argument?

Reply:
Added comments and constants.